javascript.md

```diff
@@ -0,0 +1,793 @@
+# JavaScript Guide
+
+This style guide aims to provide the ground rules for CFPB's JavaScript code, such that it's highly readable and consistent across different developers on a team. 
+
+
+## Table of Contents
+
+1. [Linting](#linting)
+1. [Modules](#modules)
+1. [Strict Mode](#strict-mode)
+1. [Spacing](#spacing)
+1. [Naming](#naming)
+1. [Semicolons](#semicolons)
+1. [Strings](#strings)
+1. [Variable Declaration](#variable-declaration)
+1. [Conditionals](#conditionals)
+1. [Equality](#equality)
+1. [Ternary Operators](#ternary-operators)
+1. [Functions](#functions)
+1. [Prototypes](#prototypes)
+1. [Object Literals](#object-literals)
+1. [Array Literals](#array-literals)
+1. [Regular Expressions](#regular-expressions)
+1. [`console` Statements](#console-statements)
+1. [Comments](#comments)
+1. [Variable Naming](#variable-naming)
+1. [Polyfills](#polyfills)
+1. [Everyday Tricks](#everyday-tricks)
+1. [Credits](#credits)
+
+## Linting
+
+Lint your code using ESLint. [CFPB-tailored ESLint options](https://github.com/cfpb/front-end/js-standards/.eslintrc) exist.
+
+## Modules
+
+This style guide assumes you're using a module system such as [CommonJS][1], [AMD][2], [ES6 Modules][3], or any other kind of module system. Modules systems provide individual scoping, avoid leaks to the `global` object, and improve code base organization by **automating dependency graph generation**, instead of having to resort to manually creating tens of `<script>` tags.
+
+Module systems also provide us with dependency injection patterns, which are crucial when it comes to testing individual components in isolation.
+
+## Strict Mode
+
+**Always** put [`'use strict';`][4] at the top of your modules. Strict mode allows you to catch nonsensical behavior, discourages poor practices, and _is faster_ because it allows compilers to make certain assumptions about your code.
+
+## Spacing
+
+Spacing must be consistent across every file in the application. To this end, using something like [`.editorconfig`][5] configuration files is highly encouraged. Here are the defaults we suggest to get started with JavaScript indentation.
+
+```ini
+# editorconfig.org
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+```
+
+We recommend using 2 spaces for indentation. The `.editorconfig` file can take care of that for us and everyone would be able to create the correct tabs by pressing the tab key.
+
+Spacing doesn't just entail tabbing, but also the spaces before, after, and in between arguments of a function declaration. This kind of spacing is typically highly irrelevant to get right, and it'll be hard for most teams to even arrive at a scheme that will satisfy everyone.
```

Comment:
I'm unclear on the purpose of this paragraph. A standards guide should probably leave out copy that says essentially "we can't offer guidance, it's un-standardizable."